level_5/lexer.hpp

```diff
@@ -0,0 +1,157 @@
+        /// which does not have a *int* type (and also friends)
+        ///
+        /// \param token_type Type of the operator
+        constexpr Token(Type token_type) : token_type_{token_type} {}
+        constexpr Token() {}
+        constexpr Token(const Token& token) {
+                token_type_= token.token_type_;
+                value_= token.value_;
+        }
+        constexpr Token& operator=(const Token& token) {
+                token_type_= token.token_type_;
+                value_= token.value_;
+                return *this;
+        }
+        constexpr Type get_token_type() const noexcept { return token_type_; }
+        constexpr int get_integer_literal() const noexcept { return value_; }
+};
+
+class Lexer {
+      public:
+        /// Lexer constructor
+        ///
+        /// \param text Raw calculator expression
+        explicit Lexer(const std::string& text)
+            : text_{text}, current_char_{text_.at(pos_)} {}
+        explicit Lexer(std::string&& text)
+            : text_{std::move(text)}, current_char_{text_.at(pos_)} {}
+
+        /// Lexucal analyzer(or scanner or token generator)
+        ///
+        /// This method is responsible for breaking the input expression into
+        /// tokens. One token at a time. \return Current Token of the input
+        /// string
+        constexpr Token get_next_token() {
+                while(current_char_ != '\0') {
+                        Type char_type= get_type_from_char(current_char_);
+                        switch(char_type) {
+                        case Type::Space:
+                                skip_whitespace();
+                                continue;
+                        case Type::Integer:
+                                return Token(Type::Integer, integet());
+                        case Type::Div:
+                                advance();
+                                return Token(Type::Div);
+                        case Type::Mul:
+                                advance();
+                                return Token(Type::Mul);
+                        case Type::Lparan:
+                                advance();
+                                return Token(Type::Lparan);
+                        case Type::Rparan:
+                                advance();
+                                return Token(Type::Rparan);
+                        case Type::Plus:
+                                advance();
+                                return Token(Type::Plus);
+                        case Type::Minus:
+                                advance();
+                                return Token(Type::Minus);
+                        default:
+                                error();
+                        }
+                }
+                return Token(Type::End, '\0');
+        }
+
+      private:
+        std::string text_;
+        std::size_t pos_{};
+        char current_char_;
```

Comment:
```suggestion
        char current_char_{};
```
maybe like this? (just like pos_)